src/Fmt_ast.ml

```diff
@@ -130,15 +127,29 @@ let function_indent c ~ctx ~default =
   | `Always, _ | _, (Top | Sig _ | Str _) -> c.conf.function_indent
   | _ -> default
 
-let fmt_expressions c width sub_exp exprs fmt fmt_expr =
+let fmt_expressions c width sub_exp exprs fmt_expr sep_before
+    sep_after_non_final sep_after_final =
```

Comment:
I think this function could take a `Params.elements_collection` argument instead.
It is always called like:
```ocaml
( p.box
    (fmt_expressions c ...
       p.sep_before p.sep_after_non_final p.sep_after_final)
```

Reply:
done